plugins/in_exec/in_exec.c

```diff
@@ -119,15 +207,40 @@ static int in_exec_config_read(struct flb_in_exec_config *exec_config,
 {
     char *cmd = NULL;
     char *pval = NULL;
+    char *follow = NULL;
+    int argc = 0;
 
     /* filepath setting */
     cmd = flb_input_get_property("command", in);
     if (cmd == NULL) {
         flb_error("[in_exec] no input 'command' was given");
         return -1;
     }
-    exec_config->cmd = cmd;
 
+    /* split into an argv array */
+    char *arg = NULL;
+    for (arg = strtok_r(cmd, " ", &cmd); arg != NULL; arg = strtok_r(NULL, " ", &cmd)) {
+        exec_config->argv = flb_realloc(exec_config->argv, (argc + 2) * sizeof(char *));
+        exec_config->argv[argc++] = arg;
+    }
+    exec_config->argv[argc] = NULL;
+
+    /* follow setting */
+    follow = flb_input_get_property("follow", in);
```

Comment:
Refer to in_exec of fluentd, `run_interval` option means `follow` mode or not.
https://docs.fluentd.org/v1.0/articles/in_exec#run_interval

I means, 
I think we don't need new option "follow".

If `Interval_Sec` and `Interval_NSec` are not set, `exec_config->follow` is FLB_TRUE. (= run_interval is not set)
if `Interval_Sec` or/and `Interval_NSec` is/are set, `exec_config->follow` is FLB_FALSE. (= run_interval is set)
Default value of `Interval_Sec` and `Interval_NSec` are 0.

Reply:
Note: We support fd event based API `flb_input_set_collector_event`. (e.g. in_kmsg uses this API)

I think the follow mode is not time based mode. How about trying to use it?

```c
flb_input_set_collector_event(in,
                                           in_exec_collect,
                                           fileno(exec_config->cmdp),
                                           config);
```